libcloud/common/aws.py

```diff
@@ -347,14 +347,15 @@ def _get_credential_scope(self, dt):
         )
 
     def _get_signed_headers(self, headers):
-        return ";".join([k.lower() for k in sorted(headers.keys())])
+        return ";".join([k.lower()
```

Comment:
I would like to understand reasoning behind this and make sure it doesn't break the generated signature. Can you please provide more context and confirm that?

Reply:
Lets ask @murlock 
Apparently without this change, the S3 Storage in Scaleway does not work.

Reply:
I would hope we have tests for various edge cases for those signature classes, but I would need to dig in.

But yeah, the docs do seem to claim the headers need to be lowecase - https://docs.aws.amazon.com/general/latest/gr/sigv4-create-canonical-request.html.

Would be good to add some test cases which confirm that.

I'll also quickly test this change with S3 when I get a chance, just to confirm this change doesn't break something.

Reply:
I did a very quick and non extensive test with the S3 driver with those code changes and it seems to work fine.

But yeah, I didn't really dig in much.

Reply:
Hi everyone
We use only lower case header name:
The data to check signature are setup at https://github.com/openstack-archive/swift3/blob/v1.11/swift3/request.py#L251
and since there is mismatch while preparing data at https://github.com/openstack-archive/swift3/blob/53614c2da837a9a53016b7d3e9911cf96ab39c77/swift3/request.py#L255 a SignatureMismatch is raised